Sources/StreamChatUI/Utils/VideoPreviewLoader.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import AVKit
+import StreamChat
+import UIKit
+
+public protocol VideoPreviewLoader: AnyObject {
+    func loadPreviewForVideo(at url: URL, completion: @escaping (Result<UIImage, Error>) -> Void)
+}
+
+final class DefaultVideoPreviewLoader: VideoPreviewLoader {
```

Comment:
Perhaps this file can be of use to you? https://github.com/GetStream/stream-chat-swift/blob/main_v2/Sources/UI/Extensions/URL+Media.swift

It has a simpler implementation of this

Reply:
It would be if we don't show previews for **remote** assets in a message list 😄 We still can reuse it switching to the background to get a preview but it's exactly what `generateCGImagesAsynchronously` does